api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/instance/InstanceRetrievalService.java

```diff
@@ -139,30 +138,32 @@ private Applications extractApplications(String responseBody) {
      * @param eurekaServiceInstanceRequest information used to query the discovery service
      * @return ResponseEntity<String> query response
      */
-    private String queryDiscoveryForInstances(EurekaServiceInstanceRequest eurekaServiceInstanceRequest) throws IOException, ParseException {
+    private String queryDiscoveryForInstances(EurekaServiceInstanceRequest eurekaServiceInstanceRequest) throws IOException {
         HttpGet httpGet = new HttpGet(eurekaServiceInstanceRequest.getEurekaRequestUrl());
         for (Header header : createRequestHeader(eurekaServiceInstanceRequest)) {
             httpGet.setHeader(header);
         }
-        CloseableHttpResponse response = httpClient.execute(httpGet);
-        final int statusCode = response.getCode();
-        final HttpEntity responseEntity = response.getEntity();
-        String responseBody = "";
-        if (responseEntity != null) {
-            responseBody = EntityUtils.toString(responseEntity, StandardCharsets.UTF_8);
-        }
-        if (statusCode >= HttpStatus.SC_OK && statusCode < HttpStatus.SC_MULTIPLE_CHOICES) {
-            return responseBody;
-        }
 
-        apimlLog.log("org.zowe.apiml.apicatalog.serviceRetrievalRequestFailed",
-            eurekaServiceInstanceRequest.getServiceId(),
-            eurekaServiceInstanceRequest.getEurekaRequestUrl(),
-            statusCode,
-            response.getReasonPhrase() != null ? response.getReasonPhrase() : responseBody
+        return httpClient.execute(httpGet, response -> {
+            final int statusCode = response.getCode();
+            final HttpEntity responseEntity = response.getEntity();
+            String responseBody = "";
+            if (responseEntity != null) {
+                responseBody = EntityUtils.toString(responseEntity, StandardCharsets.UTF_8);
+            }
```

Comment:
I don't think this will improve readability. The responseBody string is return only if the response code in success, so an additional check must be here anyway.